repose-aggregator/core/repose-core/src/test/scala/org/openrepose/core/opentracing/ReposeTracerTest.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.core.opentracing
+
+import io.opentracing.noop.{NoopTracer, NoopTracerFactory}
+import io.opentracing.propagation.Format
+import io.opentracing.util.GlobalTracer
+import io.opentracing.{ScopeManager, Span, SpanContext, Tracer}
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{FunSpec, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class ReposeTracerTest extends FunSpec with Matchers with MockitoSugar {
+
+  val testTracer = new TestTracer
+
+  describe("Repose Tracer") {
+
+    it("should register with the Global Tracer during initialization") {
+      GlobalTracer.get().isInstanceOf[ReposeTracer.type]
+    }
+
+    it("should initialize with a NoOp Tracer") {
+      ReposeTracer.get().isInstanceOf[NoopTracer]
+    }
+
+    it("should initialize not registered") {
+      ReposeTracer.isRegistered shouldBe false
+    }
+
+    it("should fail to register a <null> Tracer") {
+      val exception = intercept[NullPointerException] {
+        ReposeTracer.register(null)
+      }
+      exception.getMessage shouldBe "Cannot register ReposeTracer <null>."
+    }
+
+    it("should allow the registration a Tracer") {
+      ReposeTracer.register(testTracer)
+      ReposeTracer.get should equal(testTracer)
+      ReposeTracer.isRegistered shouldBe true
+    }
+
+    it("should allow the registration a different Tracer") {
+      val testTracerToo = new TestTracer
+      ReposeTracer.register(testTracerToo)
+      testTracer shouldNot equal(testTracerToo)
```

Comment:
Not sure what the point in this is? testing that two objects you newed up aren't the same? I think you are trying to make sure that you can swap it, but this assertion doesn't do that, and the test itself doesn't either without depending on prior tests already running.

Reply:
That specific line is to ensure the two Tracers are in fact not the same. The next line is what ensures the new Tracer was registered. I'll add in another line to make sure the first Tracer was added.